### ISSUE TITLE
Issues in t/translator.t

### ISSUE THREAD

**Issue by matsduf** [zonemaster/zonemaster-engine#817]
This is run on develop branch on commit 81fc8abe4cbf2aa3a12d30f6d266a32c2c08ed07 and on FreeBSD. 

There are two issues with the unit test:
1. It always says "Reference found where even-sized list expected", which does not sounds like the expected behavior.
2. It fails if the locale is set to a language that we have translation for, here exemplified with French.

LANG unset -- PASS, but with Perl code warning:
```
$ env | grep -i utf
$ PERL_DL_NONLAZY=1 "/usr/local/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/translator.t 
t/translator.t .. 1/? Reference found where even-sized list expected at /usr/home/matsd/.cpanm/work/1603398129.17914/Zonemaster-Engine-v3.1.2/blib/lib/Zonemaster/Engine/Translator.pm line 121.
t/translator.t .. ok   
All tests successful.
Files=1, Tests=6,  1 wallclock secs ( 0.03 usr  0.00 sys +  0.73 cusr  0.12 csys =  0.89 CPU)
Result: PASS
```
LANG set -- FAIL plus the same Perl code warning:
```
$ env | grep -i utf
LANG=fr_FR.UTF-8
$ PERL_DL_NONLAZY=1 "/usr/local/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'inc', 'blib/lib', 'blib/arch')" t/translator.t 
t/translator.t .. 1/? Reference found where even-sized list expected at /usr/home/matsd/.cpanm/work/1603398129.17914/Zonemaster-Engine-v3.1.2/blib/lib/Zonemaster/Engine/Translator.pm line 121.

#   Failed test 'string to_stringd as expected'
#   at t/translator.t line 30.
#                   '   0.08 INFO      Une zone parente 'nothing.nowhere' a pu être trouvée pour le nom de domaine testé.'
#     doesn't match '(?^:   0.\d\d INFO      Parent domain 'nothing.nowhere' was found for the tested domain.)'
# Looks like you failed 1 test of 6.
t/translator.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/6 subtests 

Test Summary Report
-------------------
t/translator.t (Wstat: 256 Tests: 6 Failed: 1)
  Failed test:  5
  Non-zero exit status: 1
Files=1, Tests=6,  1 wallclock secs ( 0.04 usr  0.00 sys +  0.75 cusr  0.12 csys =  0.91 CPU)
Result: FAIL
Failed 1/1 test programs. 1/6 subtests failed.
```

**Comment by matsduf** [comment]
@vlevigneron, I put you on this issue. Can you have a look at it?

**Comment by matsduf** [comment]
If you go back to commit be87249, just before #805 was merged, neither of the issues seems to be there.

**Comment by matsduf** [comment]
Resolved by #826.
